plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java

```diff
@@ -193,17 +193,25 @@ public TlsGrpcQueryClientFactory(PinotGrpcServerQueryClientConfig grpcClientConf
         {
             requireNonNull(grpcClientConfig, "grpcClientConfig is null");
             requireNonNull(tlsConfig, "tlsConfig is null");
-            this.config = new GrpcQueryClient.Config(ImmutableMap.<String, Object>builder()
+            ImmutableMap.Builder<String, Object> tlsConfigBuilder = ImmutableMap.<String, Object>builder()
                     .put(CONFIG_MAX_INBOUND_MESSAGE_BYTES_SIZE, String.valueOf(grpcClientConfig.getMaxInboundMessageSize().toBytes()))
-                    .put(CONFIG_USE_PLAIN_TEXT, String.valueOf(grpcClientConfig.isUsePlainText()))
-                    .put(GRPC_TLS_PREFIX + "." + KEYSTORE_TYPE, tlsConfig.getKeystoreType())
-                    .put(GRPC_TLS_PREFIX + "." + KEYSTORE_PATH, tlsConfig.getKeystorePath())
-                    .put(GRPC_TLS_PREFIX + "." + KEYSTORE_PASSWORD, tlsConfig.getKeystorePassword())
-                    .put(GRPC_TLS_PREFIX + "." + TRUSTSTORE_TYPE, tlsConfig.getTruststoreType())
-                    .put(GRPC_TLS_PREFIX + "." + TRUSTSTORE_PATH, tlsConfig.getTruststorePath())
-                    .put(GRPC_TLS_PREFIX + "." + TRUSTSTORE_PASSWORD, tlsConfig.getTruststorePassword())
-                    .put(GRPC_TLS_PREFIX + "." + SSL_PROVIDER, tlsConfig.getSslProvider())
-                    .buildOrThrow());
+                    .put(CONFIG_USE_PLAIN_TEXT, String.valueOf(grpcClientConfig.isUsePlainText()));
+            setProperty(tlsConfigBuilder, GRPC_TLS_PREFIX + "." + KEYSTORE_TYPE, tlsConfig.getKeystoreType());
+            setProperty(tlsConfigBuilder, GRPC_TLS_PREFIX + "." + KEYSTORE_PATH, tlsConfig.getKeystorePath());
+            setProperty(tlsConfigBuilder, GRPC_TLS_PREFIX + "." + KEYSTORE_PASSWORD, tlsConfig.getKeystorePassword());
+            setProperty(tlsConfigBuilder, GRPC_TLS_PREFIX + "." + TRUSTSTORE_TYPE, tlsConfig.getTruststoreType());
+            setProperty(tlsConfigBuilder, GRPC_TLS_PREFIX + "." + TRUSTSTORE_PATH, tlsConfig.getTruststorePath());
+            setProperty(tlsConfigBuilder, GRPC_TLS_PREFIX + "." + TRUSTSTORE_PASSWORD, tlsConfig.getTruststorePassword());
+            setProperty(tlsConfigBuilder, GRPC_TLS_PREFIX + "." + SSL_PROVIDER, tlsConfig.getSslProvider());
+            this.config = new GrpcQueryClient.Config(tlsConfigBuilder.buildOrThrow());
```

Comment:
Could you explain why these change are needed in this PR?

Reply:
The issue is Pinot Config does not like null as the value, and throws exceptions when initializing grpc client, hence, we need to ignore the property instead of setting it as null.

Reply:
I know it. I'm asking "why these change are needed in this PR". Do you mean this change relates to https commit?

Reply:
The reason is when enable TLS in Pinot for HTTPS and grpc together. The initialization for the pinot grpc client will fail. So I cannot perform the whole e2e testing.

It's also possible to fix this as a separated PR, but I don't feel like it's worth it, cause typical users won't enable tls for grpc but leave controller/broker non-tls.

Reply:
Thanks, I get it now. First of all, we should use `Optional` type in the config properties instead of nullable `String`. Please left a TODO comment.